pkg/minikube/machine/filesync_test.go

```diff
@@ -122,17 +102,24 @@ func TestAssetsFromDir(t *testing.T) {
 					return err
 				}()
 				if err != nil {
-					t.Errorf("unable to create file on fs: %v", err)
-					return
+					t.Fatalf("unable to create file on fs: %v", err)
 				}
 			}
 
 			actualFiles, err := assetsFromDir(testFileBaseDir, test.vmPath, test.flatten)
 			if err != nil {
-				t.Errorf("got unexpected error adding minikube dir assets: %v", err)
-				return
+				t.Fatalf("got unexpected error adding minikube dir assets: %v", err)
```

Comment:
It this test fail here, the cleanup will not run since we call it only after this failure.